Modules/_datetimemodule.c

```diff
@@ -988,7 +1100,7 @@ new_date_ex(int year, int month, int day, PyTypeObject *type)
 }
 
 #define new_date(year, month, day) \
-    new_date_ex(year, month, day, get_datetime_state()->date_type)
+    new_date_ex(year, month, day, DATE_TYPE(NO_STATE))
```

Comment:
The macros I've added for the various static objects don't need to be passed a state, but I found the consistency of passing one to be more clear and added `NO_STATE` to make the special case obvious.

Reply:
done